sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -220,7 +220,7 @@ export function createTokenCycler(
     }
 
     if (cycler.shouldRefresh) {
-      refresh(scopes, tokenOptions);
+      await refresh(scopes, tokenOptions);
```

Comment:
The omitted `await` might be intentional as the current token has not expired yet.

Reply:
Actuall we just return Promises from this method so this is expected. We probably can remove the `async`.